backend/dataall/modules/metadata_forms/db/metadata_form_models.py

```diff
@@ -0,0 +1,149 @@
+    fieldUri = Column(String, ForeignKey('metadata_form_field.uri'), primary_key=True)
+    type = Column(String, nullable=False)
+    field = relationship('MetadataFormField', backref='attached_fields')
+
+    __table_args__ = (PrimaryKeyConstraint('attachedFormUri', 'fieldUri'),)
+    __mapper_args__ = {'polymorphic_identity': 'attached_metadata_form_field', 'polymorphic_on': type}
+
+    @property
+    def value(self):
+        raise NotImplementedError('Basic AttachedMetadataFormField has no implemented property value')
+
+    @validates('type')
+    def update_type(self, key, new_type):
+        if new_type != self.field.type:
+            raise ValueError("Value type doesn't match field type")
+
+
+class StringAttachedMetadataFormField(AttachedMetadataFormField):
+    __tablename__ = 'string_attached_metadata_form_field'
+    attachedFormUri = Column(String, primary_key=True)
+    fieldUri = Column(String, primary_key=True)
+    sValue = Column(String, nullable=False)
+    __mapper_args__ = {'polymorphic_identity': MetadataFormFieldType.String}
+
+    __table_args__ = (
+        ForeignKeyConstraint(
+            ['attachedFormUri', 'fieldUri'],
+            ['attached_metadata_form_field.attachedFormUri', 'attached_metadata_form_field.fieldUri'],
+        ),
+    )
+
+    @property
+    def value(self):
+        return self.sValue
+
+
+class BooleanAttachedMetadataFormField(AttachedMetadataFormField):
+    __tablename__ = 'boolean_attached_metadata_form_field'
+    attachedFormUri = Column(String, primary_key=True)
+    fieldUri = Column(String, primary_key=True)
+    bValue = Column(Boolean, nullable=False)
+    __mapper_args__ = {'polymorphic_identity': 'Boolean'}
+
+    __table_args__ = (
+        ForeignKeyConstraint(
+            ['attachedFormUri', 'fieldUri'],
+            ['attached_metadata_form_field.attachedFormUri', 'attached_metadata_form_field.fieldUri'],
+        ),
+    )
+
+    @property
+    def value(self):
+        return self.bValue
+
+
+class IntegerAttachedMetadataFormField(AttachedMetadataFormField):
+    __tablename__ = 'integer_attached_metadata_form_field'
+    attachedFormUri = Column(String, primary_key=True)
+    fieldUri = Column(String, primary_key=True)
+    iValue = Column(Integer, nullable=False)
+    __mapper_args__ = {'polymorphic_identity': 'Integer'}
+
+    __table_args__ = (
+        ForeignKeyConstraint(
+            ['attachedFormUri', 'fieldUri'],
+            ['attached_metadata_form_field.attachedFormUri', 'attached_metadata_form_field.fieldUri'],
+        ),
+    )
+
+    @property
+    def value(self):
+        return self.iValue
+
+
+class GlossaryTermAttachedMetadataFormField(AttachedMetadataFormField):
+    __tablename__ = 'glossary_term_attached_metadata_form_field'
+    attachedFormUri = Column(String, primary_key=True)
+    fieldUri = Column(String, primary_key=True)
+    gtValue = Column(String, nullable=False)
+    __mapper_args__ = {'polymorphic_identity': 'Glossary Term'}
```

Comment:
can we pass `MetadataFormFieldType.GlossaryTerm` instead of hardcoded` 'GlossaryTerm'`

Reply:
[dbb54e7](https://github.com/data-dot-all/dataall/pull/1422/commits/dbb54e76b22c0c987dc115eb749c9b89c6a11ae7)